hnn_core/network.py

```diff
@@ -870,17 +870,37 @@ def _instantiate_drives(self, n_trials=1):
                 event_times = list()  # new list for each trial and drive
 
                 # loop over drive 'cells' and create event times for each
-                for this_cell_drive_conn in drive['conn'].values():
-                    for drive_cell_gid in this_cell_drive_conn['src_gids']:
-                        event_times.append(_drive_cell_event_times(
-                            drive['type'], this_cell_drive_conn,
-                            drive['dynamics'], trial_idx=trial_idx,
-                            drive_cell_gid=drive_cell_gid,
-                            seedcore=drive['seedcore'])
-                        )
-                    # only create one event_times list for globals
-                    if not drive['cell_specific']:
-                        break  # loop over drive['conn'].values
+                if drive['cell_specific']:
+                    for this_cell_drive_conn in drive['conn'].values():
+                        for drive_cell_gid in this_cell_drive_conn['src_gids']:
+                            event_times.append(_drive_cell_event_times(
+                                drive['type'], this_cell_drive_conn,
+                                drive['dynamics'], trial_idx=trial_idx,
+                                drive_cell_gid=drive_cell_gid,
+                                seedcore=drive['seedcore'])
+                            )
+                else:
+                    for this_cell_drive_conn in drive['conn'].values():
+                        has_weight = np.logical_or(
+                            this_cell_drive_conn['ampa'],
+                            this_cell_drive_conn['nmda'])
```

Comment:
isn't this logic already in `_drive_cell_event_times` ?

https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/drives.py#L237

Reply:
Ah good point. I think both are necessary. For `cell_specific=True`, this returns an empty list for cells with weight=0.

However, for `cell_specific=False`, you need to be careful because all cells are sharing the same event times. Sometimes the parameters of the drives lead to an empty list, despite the cell weight being non-zero. This ensures that if an empty list is returned, it is due to the simulation parameters and not the weights.

Reply:
Said another way, the loop exits the first time ` _drive_cell_event_times` is called so that there is only one set of event_times. You need to be sure the function receives a drive_conn with positive weights.